Appc 1051 develop billing standalone sdk

brunovarelaAptoide opened the pull request:
**What does this PR do?**
This PR transform billing sdk into a standalone application. The changes were:
-Removed dependencies
-Added a new Webservice connection to obtain skuDetails
-Dummy invocations in case of the user don't have a Wallet Installed.
-New JSONParser.Used the Android one org.json
-Launch the wallet install popup when the user launches billing flow and doesn't have wallet installed.
-Added new web service to check if the billing is supported in the app.
-Correction of a bug in Key validation

**Database changed?**
No

**Where should the reviewer start?**
-CatapultAppcoinsBilling.java
-AppcoinsBillingClient.java
-AndroidBillingMapper.java
-GetSkuDetailsService.java
-WSServiceController.java
-IsBillingSupportedService.java

**How should this be manually tested?**
 Using the toolbox application.Comment the appcoins-ads code to insure the wallet install pop up doesn't appear.

**What are the relevant tickets?**
APPC 1051

**Questions:**

   Does this add new dependencies which need to be added? No




**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

brunovarelaAptoide committed:
Wallet verification

- Added wallet verfication to consume, startService,getPurchases
-Removed dependencies

brunovarelaAptoide committed:
skuDetails web service

-New WebService to check the skuDetails

brunovarelaAptoide committed:
Pop up on Purchase Intent

- Pop up only when purchase intent is pressed
- Service corrections

brunovarelaAptoide committed:
Mapper v 0.1

- early mapper version

brunovarelaAptoide committed:
New WS and Parser

-New ws integrated
-Parser of json for that WS

brunovarelaAptoide committed:
Merge with new JSONParser

-New JsonParser

nunomonteiro commented on android-appcoins-billing/build.gradle:
```diff
@@ -1,75 +1,89 @@
+  buildTypes {
+    debug {
+      minifyEnabled false
+      buildConfigField 'String', 'BDS_WALLET_PACKAGE_NAME',
```
Shouldn't the dev package build with the dev wallet name?

nunomonteiro commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/CatapultAppcoinsBilling.java:
```diff
@@ -17,10 +19,11 @@
 import com.appcoins.sdk.billing.SkuDetailsResponseListener;
 
-public class CatapultAppcoinsBilling implements AppcoinsBillingClient {
+public class CatapultAppcoinsBilling implements AppcoinsBillingClient , DialogVisibleListener {
```
I'm gonna be picky but these names matter to give the user a sense of the class purpose.
I know what you're trying to say but it's ambiguous. Does it listen to the dialog only when it's visible? I went ahead and saw that the answer is no. But the name should tell me this already.
I suggest changing it to IsDialogVisibleListerner or even better DialogVisibilityListener (you'll see why on the next comment)

nunomonteiro commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/CatapultAppcoinsBilling.java:
```diff
@@ -88,6 +99,10 @@ public void consumeAsync(String token, ConsumeResponseListener consumeResponseLi
     return billing.isReady();
   }
+
+  @Override public void OnDialogVisibleListener(boolean value) {
```
this function should be called onDialogVisibilityChanged. This way you know it's supposed to be called every time the visibility changes and not just when the dialog is visible. Also notice the lowercase o! Java methods should always start with lowercase letter.

nunomonteiro commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/helpers/AndroidBillingMapper.java:
```diff
@@ -35,56 +32,64 @@ public PurchasesResult mapPurchases(Bundle bundle, String skuType) {
+            developerPayload = jsonElement.getString("developerPayload");
+          }
+        } catch (org.json.JSONException e) {
+
```
Avoid empty catch statements at all costs! At least always log the exception with an error. If it's not your code, check with the last person to have committed there what an exception there might mean and treat it accordingly.

nunomonteiro commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/helpers/AndroidBillingMapper.java:
```diff
@@ -35,56 +32,64 @@ public PurchasesResult mapPurchases(Bundle bundle, String skuType) {
+            token = jsonElement.getString("token");
+          }
+        } catch (org.json.JSONException e) {
+
```
Same here

nunomonteiro commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/helpers/AndroidBillingMapper.java:
```diff
@@ -35,56 +32,64 @@ public PurchasesResult mapPurchases(Bundle bundle, String skuType) {
+            isAutoRenewing = jsonElement.getBoolean("autoRenewing");
+          }
+        } catch (org.json.JSONException e) {
+
```
same here

nunomonteiro commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/helpers/AndroidBillingMapper.java:
```diff
@@ -35,56 +32,64 @@ public PurchasesResult mapPurchases(Bundle bundle, String skuType) {
+            new Purchase(id, skuType, purchaseData, decodedSignature, purchaseTime, purchaseState,
+                developerPayload, token, packageName, sku, isAutoRenewing));
+      } catch (JSONException e) {
+        e.printStackTrace();
```
This would be a good start

nunomonteiro commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/helpers/DialogVisibleListener.java:
```diff
@@ -0,0 +1,6 @@
+package com.appcoins.sdk.android_appcoins_billing.helpers;
+
+public interface DialogVisibleListener {
```
Same changes here:
filename
interface name
method name

nunomonteiro commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/helpers/PayloadHelper.java:
```diff
@@ -30,8 +29,8 @@
    */
-  public static String buildIntentPayload(@Nullable String orderReference,
-      @Nullable String developerPayload, String origin) {
+  public static String buildIntentPayload(final String orderReference,
```
Will this always get arguments different from null now? This might require updates to documentation

nunomonteiro commented on appcoins-billing/src/main/java/com/appcoins/sdk/billing/GetSkuDetailsService.java:
```diff
@@ -0,0 +1,65 @@
+  }
+
+  private String buildURL(String packageName, List<String> sku) {
+    String url = serviceUrl + "/inapp/8.20180518/packages/packageName/products?names=".replaceFirst(
```
the hardcoded path should be a final at the top of the file for better maintainability of the code. If someone else would need to change the path they wouldn't have to fish that var in the middle of the file.

nunomonteiro commented on toolbox/src/main/java/com/asf/appcoins/toolbox/Application.java:
```diff
@@ -1,25 +1,23 @@
     public final static String developerAddress = "0x75e8BdCf84B9bd585753dB64A8637A52Bd46f851";
 
-    private static AppCoinsAds adsSdk;
+    //private static AppCoinsAds adsSdk;
```
Check with the Wallet team how to deal with the POA being present on the toolbox

nunomonteiro reviewed (changes requested):
@brunovarelaAptoide 
Parece muito mas é mais texto e explicaçoes para o que tou a dizer de forma a perceberes o porque das coisas e ser mais fácil evitares o erro da próxima vez.
@JoaoRaimundo @Trinkes 
Primeira tentativa de eu agir como intermediário entre os pulls requests da equipa e vcs.
Sugiro que em vez de verem as changes, vejam só o feedback que eu dei e o validem. 
Se tiver tudo ok, mantemos este esquema e as cenas seguem para vcs após terem levado uma vista d'olhos minha. Boa?

brunovarelaAptoide committed:
Changes Code Review

-Changes according to the code review

brunovarelaAptoide committed:
Changes Scenario 1

- Added a new Webservice to check if app supports billing.
- Removed a bug in key validation
- refactor the package name in appcoins billing
- Created a controller for WS services
- Created a new listenner for isBillingSupported operation

brunovarelaAptoide committed:
Publishing added

changes in android manifest for publishing.

brunovarelaAptoide committed:
Patch changes

Applied patch for publishing.

brunovarelaAptoide committed:
Patch changes

brunovarelaAptoide committed:
Names changed

Changed name on AndroidManifest and all the consequent places.

brunovarelaAptoide committed:
- Creation of a Dummy Stub

- Creation of a Dummy to make the no Wallet operations.
- Removed arguments from all Parameters in WalletUtils.
- Added a function to retrieve the activity active in WalletUtils
- Validation checks for No Wallet Scenario
- Removed Service for billing is supported.

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java:
```diff
@@ -0,0 +1,64 @@
+import com.appcoins.sdk.billing.WSServiceController;
+import java.util.ArrayList;
+
+public class AppcoinsBillingStubHelper extends android.os.Binder implements AppcoinsBilling {
```
Why is this class extending Binder?

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java:
```diff
@@ -0,0 +1,64 @@
+  }
+
+  @Override public int consumePurchase(int apiVersion, String packageName, String purchaseToken) {
+    return ResponseCode.OK.getValue();
```
since we are not consuming the purchase, we should answer with an error instead of the ok

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java:
```diff
@@ -0,0 +1,64 @@
+
+  public static abstract class Stub {
+
+    public static AppcoinsBilling AsInterface(IBinder service) {
```
the method name shouldn't start with a capital letter

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppCoinsAndroidBillingRepository.java:
```diff
@@ -1,51 +1,54 @@
   private final int apiVersion;
   private final String packageName;
-  private final AndroidBillingMapper billingMapper;
+  private final Context context;
```
Context is not used

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppCoinsAndroidBillingRepository.java:
```diff
@@ -1,51 +1,54 @@
-package com.appcoins.sdk.android_appcoins_billing.helpers;
+package com.appcoins.sdk.billing.helpers;
 
+import android.app.Activity;
```
not used

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppCoinsAndroidBillingRepository.java:
```diff
@@ -1,51 +1,54 @@
     this.apiVersion = apiVersion;
     this.packageName = packageName;
-    this.billingMapper = billingMapper;
+    this.context = context;
```
remove not used context

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/PayloadHelper.java:
```diff
@@ -1,12 +1,11 @@
-package com.appcoins.sdk.android_appcoins_billing.helpers;
+package com.appcoins.sdk.billing.helpers;
```
why did you change PayloadHelper?

Trinkes commented on android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/WalletUtils.java:
```diff
@@ -0,0 +1,97 @@
+
+  public static Context context;
+
+  public static Activity act;
```
no need for holding a static reference for the activity anymore

Trinkes commented on scripts/publish.sh:
```diff
@@ -1,8 +1,8 @@
 cd ..
 # Add '.dev' on the version name. Non dev builds are done though jenkins build only
-#export BUILD_NAME_SUFFIX=".dev"
+export BUILD_NAME_SUFFIX=".dev"
```
do we want do commit this?

brunovarelaAptoide committed:
Pull Request Changes

- Changes requested on pull request.
- Wallet install prompt in dummy stub.

brunovarelaAptoide committed:
Changes

- Removed unusued argument

brunovarelaAptoide committed:
Reconnect to AIDL after wallet install

- Added a wait() and notify routines in StubHelper.
-Toolbox change to invoke Billing functions in new threads
-Wallet utils changed to invoke the popup in ui main thread.

brunovarelaAptoide committed:
PR Changes

- Changes Exception
- To keep coerence in WalletUtils remove RunOnUIThread, and placed it in Stub

Trinkes commented on appcoins-billing/src/main/java/com/appcoins/sdk/billing/WSServiceController.java:
```diff
@@ -0,0 +1,17 @@
+
+  private static String wsUrl;
+
+  public static void setWsUrl(String url) {
```
remove this setter and use the build config value directly here

brunovarelaAptoide committed:
Changes Pull Request

- exception handling
- stop using a repository connection
- sycronized the whole block with isServiceBlock
- Changed variable to isServiceBlock
- used the sycronized tag in getPurchases and consumePurchase

brunovarelaAptoide committed:
Pull Rquest Changes

- Removed SetUrl method and added a new parameter in GetSkuDetailsService

Trinkes committed:
rename GetSkuDetailsService method to getSkuDetailsService

Trinkes committed:
remove synchronization from Sub Helper methods

Trinkes committed:
Merge branch 'APPC-1051-Develop-Billing-standalone-SDK' into APPC-1051-Behaviour-after-install-wallet

# Conflicts:
#	android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java
#	android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/CatapultBillingAppCoinsFactory.java

Trinkes committed:
Merge pull request #21 from Aptoide/APPC-1051-Behaviour-after-install-wallet

Appc 1051 behaviour after install wallet
